sam-app/hello_world/app.py

```diff
@@ -1,9 +1,152 @@
+        if "Minimum" not in event["StatisticValues"]:
+            return "Statistic Values require a Minimum."
+        try:
+            event["StatisticValues"]["Minimum"] = float(event["StatisticValues"]["Minimum"])
+        except ValueError:
+            return "Invalid Minimum"
+        if "Maximum" not in event["StatisticValues"]:
+            return "Statistic Values require a Maximum."
+        try:
+            event["StatisticValues"]["Maximum"] = float(event["StatisticValues"]["Maximum"])
+        except ValueError:
+            return "Invalid Maximum"
+        if event["StatisticValues"]["Minimum"] > event["StatisticValues"]["Maximum"]:
+            return "Minimum cannot be greater than Maximum."
+    if "Unit" in event:
+        event["Unit"] = str(event["Unit"])
+        possibleUnits = ['Seconds','Microseconds','Milliseconds','Bytes',\
+            'Kilobytes','Megabytes','Gigabytes','Terabytes','Bits','Kilobits',\
+            'Megabits','Gigabits','Terabits','Percent','Count','Bytes/Second',\
+            'Kilobytes/Second','Megabytes/Second','Gigabytes/Second',\
+            'Terabytes/Second','Bits/Second','Kilobits/Second','Megabits/Second',\
+            'Gigabits/Second','Terabits/Second','Count/Second','None']
+        if event["Unit"] not in possibleUnits:
+            return "Unit type not available"
+    if "StorageResolution" in event:
+        try:
+            event["StorageResolution"] = int(event["StorageResolution"])
+        except ValueError:
+            return "StorageResolution type must be integer."
+    return "Validated"
+
+def formMessage(event):
+    """Forms the message with the proper syntax given the user input,
+    including only those values provided by the user.
+
+    Parameters:
+        event (dict): the user input
+
+    Returns:
+        message (dict): the message with the correct format
+    """
+
+    message = {
+        "Namespace": event["Namespace"],
+        "MetricData": [
+            {
+                "MetricName": event["MetricName"]
+            },
+        ]
+    }
+    if "Dimensions" in event:
+        message["MetricData"][0]["Dimensions"] = event["Dimensions"]
+    if "Timestamp" in event:
+        message["MetricData"][0]["Timestamp"] = event["Timestamp"]
+    if "Value" in event:
+        message["MetricData"][0]["Value"] = event["Value"]
+    if "StatisticValues" in event:
+        message["MetricData"][0]["StatisticValues"] = event["StatisticValues"]
+    if "Unit" in event:
+        message["MetricData"][0]["Unit"] = event["Unit"]
+    if "StorageResolution" in event:
+        message["MetricData"][0]["StorageResolution"] = event["StorageResolution"]
+    return message
+
+def metricLogger(message):
+    """Puts the user input message into a newly created log stream.
+
+    Parameters:
+        message (dict): the message to put to the log stream
+
+    Returns:
+        None
+    """
+    client = boto3.client('logs',region_name="us-east-1")
```

Comment:
Don't put region here.

Reply:
Agreed with Wujing. Also, just statically initialize this client outside of a method rather than reinitializing it on every request. boto3 often has some caching under the covers which you won't be taking advantage of if you create a new client every request.